### PR TITLE
Adds Planning and Parsing Support for Alter Table

### PR DESCRIPTION
**Pull request by GuptaManan100**
Adds Planning and Parsing Support for `Alter Table`

### CODE REVIEW

**Review by shlomi-noach — APPROVED** [review]
Scary!

File: go/vt/vtgate/planbuilder/plan_test.go [hunk 1]
```diff
@@ -181,6 +181,8 @@ func TestPlan(t *testing.T) {
 	testFile(t, "union_cases.txt", testOutputTempDir, vschemaWrapper)
 	testFile(t, "transaction_cases.txt", testOutputTempDir, vschemaWrapper)
 	testFile(t, "lock_cases.txt", testOutputTempDir, vschemaWrapper)
+	testFile(t, "ddl_cases_no_default_keyspace.txt", testOutputTempDir, vschemaWrapper)
+	testFile(t, "show_cases_no_default_keyspace.txt", testOutputTempDir, vschemaWrapper)
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/plan_test.go:185]: Was there a need to create these two new files? The existing tests were to check even having default keyspace ddl and show queries are routes to desired keyspace.

**Review by harshit-gangal — APPROVED** [review]
LGTM overall.

File: go/vt/vtgate/executor_test.go [hunk 2]
```diff
@@ -1125,7 +1125,7 @@ func TestExecutorDDL(t *testing.T) {
 
 	stmts := []string{
 		"create table t1(id bigint primary key)",
-		"alter table t1 add primary key id",
+		"alter table t2 add primary key id",
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/vtgate/executor_test.go:1128]: Yes, the new code will find t1 as it is in Vschema and this test is for the table that is not there.

File: go/vt/vtgate/planbuilder/plan_test.go [hunk 3]
```diff
@@ -198,6 +200,11 @@ func TestSysVarSetDisabled(t *testing.T) {
 func TestOne(t *testing.T) {
 	vschema := &vschemaWrapper{
 		v: loadSchema(t, "schema_test.json"),
+		keyspace: &vindexes.Keyspace{
+			Name:    "main",
+			Sharded: false,
+		},
+		tabletType: topodatapb.TabletType_MASTER,
 	}
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/vtgate/planbuilder/plan_test.go:208]: Yes you are right

[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/vtgate/planbuilder/plan_test.go:208]: removed in the last commit

File: go/vt/vtgate/planbuilder/plan_test.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/vtgate/planbuilder/plan_test.go:185]: if the default keyspace is set, then only that is checked for tables without qualifiers but some of these tests relied on vtgate finding the table without adding the qualifier.

File: go/vt/vtgate/planbuilder/ddl.go [hunk 4]
```diff
@@ -45,18 +45,26 @@ func buildDDLPlans(sql string, ddlStatement sqlparser.DDLStatement, vschema Cont
...
 		if err != nil {
-			return nil, nil, err
+			_, isNotFound := err.(vindexes.NotFoundError)
+			if !isNotFound {
+				return nil, nil, err
+			}
 		}
-		if table == nil {
-			return nil, nil, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "table does not exists: %s", ddlStatement.GetTable().Name.String())
+		if err != nil || table == nil {
```
[the remark this answers is not in the document]
Comment (GuptaManan100) [reply go/vt/vtgate/planbuilder/ddl.go]: fixed in the last commit

**Pull request merged**